### DISCUSSION TITLE
Tab width settings doesn't take effect in markdown codeblocks and code editor

### DISCUSSION THREAD

**Discussion by so1ve** [community/community#64576] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

For example: Set tab width to `2` in your github settings, and then use the web editor to edit a file. The editor still uses 8 spaces for a tab, not 2.

**Comment by arafajs** [comment]
This is because the web editor uses a browser-based Markdown parser, which does not support the tab-size property.

To get the desired tab width in a Markdown code block, you can use the following workarounds:

- Use spaces instead of tabs. This is the simplest solution, and it will work in all Markdown parsers.
- Use fenced code blocks. Fenced code blocks are enclosed by three backticks (`) on each side. The parser will ignore any tabs within a fenced code block, and use the specified number of spaces instead.
- Use a Markdown editor that supports the tab-size property. There are a few Markdown editors that support this property, such as Visual Studio Code and Sublime Text.

**Reply by so1ve** [reply]
Thanks for your reply! I'll try your solutions. I used to use spaces instead of tabs, but soon I migrated to tabs for better accessibility.

I'm not sure if I have done anything wrong - please checkout https://github.com/so1ve/unplugin-vue-complex-types, these tabs are still rendered as 8 spaces (on mobile).

However, we should inherit code editor's tab size from user's settings.

**Reply by so1ve** [reply]
Wait… ChatGPT?
